### PR TITLE
Authentication end integration tests

### PR DESCRIPTION
**Pull request by ignaciojonas**

### CODE REVIEW

File: lib/auth0/api/authentication_endpoints.rb [hunk 1]
```diff
@@ -132,11 +133,9 @@ def phone_login(phone_number, code, scope = 'openid')
 
       # Retrives the SAML 2.0 metadata
       # @see https://auth0.com/docs/auth-api#!#get--samlp--client_id-
-      # @param client_id [string] Client id
       # @return [xml] SAML 2.0 metadata
-      def saml_metadata(client_id)
-        raise Auth0::InvalidParameter, 'Must supply a valid client_id' if client_id.to_s.empty?
-        get("/samlp/metadata/#{client_id}")
+      def saml_metadata
+        get("/samlp/metadata/#{@client_id}")
```
Comment (falvarisqueta) [inline lib/auth0/api/authentication_endpoints.rb:138]: is there any scenario that clientid can be empty with the update?

File: lib/auth0/api/authentication_endpoints.rb [hunk 2]
```diff
@@ -90,10 +90,11 @@ def change_password(email, password, connection_name = UP_AUTH)
...
         request_params = {
-          client_id:    @client_id,
-          email:        email,
-          send:         send,
-          auth_params:  auth_params
+          client_id:   @client_id,
+          connection:  'email',
+          email:       email,
+          send:        send,
+          authParams:  auth_params
```
Comment (falvarisqueta) [inline lib/auth0/api/authentication_endpoints.rb:97]: any reason why is authParams and not auth_params?

Comment (ignaciojonas) [inline lib/auth0/api/authentication_endpoints.rb:97]: In the documentation the parameter is authParams (https://auth0.com/docs/api/authentication#!#post--with_email)

File: lib/auth0/api/authentication_endpoints.rb (hunk 1 above)
Comment (ignaciojonas) [inline lib/auth0/api/authentication_endpoints.rb:138]: You cannot create a client without a client_id

**Comment by falvarisqueta** [comment]
looks good to me

**Pull request merged**